static/src/javascripts/projects/common/modules/commercial/membership-engagement-banner.js

```diff
@@ -0,0 +1,215 @@
+ * otherwise a populated params object that looks like this:
+ *
+ *  {
+ *    minArticles: 5, // how many articles should the user see before they get the engagement banner?
+ *    messageText: "..."
+ *    colourStrategy: // a function to determine what css class to use for the banner's colour
+ *    buttonCaption: "Become a Supporter"
+ *  }
+ *
+ */
+const deriveBannerParams = (location: string): Object | string => {
+    const defaultParams = membershipEngagementBannerUtils.defaultParams(
+        location
+    );
+    const userTest = getUserTest();
+    const campaignId = userTest ? userTest.campaignId : undefined;
+    const userVariant = getUserVariant(userTest);
+
+    if (userVariant && userVariant.blockEngagementBanner) {
+        return DO_NOT_RENDER_ENGAGEMENT_BANNER;
+    }
+
+    return Object.assign(
+        {},
+        defaultParams,
+        getUserVariantParams(userVariant, campaignId, defaultParams.offering)
+    );
+};
+
+// Used to send an interaction if the engagement banner is shown.
+const recordInteraction = (interaction: Object): void => {
+    if (interaction) {
+        const component = interaction.component;
+        const value = interaction.value;
+
+        if (component && value) {
+            ophan.record({
+                component,
+                value,
+            });
+        }
+    }
+};
+
+const paypalAndCreditCardImage =
+    (config.images &&
+        config.images.acquisitions &&
+        config.images.acquisitions['paypal-and-credit-card']) ||
+    '';
+
+const selectSequentiallyFrom = (array: Array<any>): any =>
+    array[local.get('gu.alreadyVisited') % array.length];
+
+const showBanner = (params: Object): void => {
+    if (params === DO_NOT_RENDER_ENGAGEMENT_BANNER || isBlocked()) {
+        return;
+    }
+
+    const colourClass = params.colourStrategy();
+
+    const messageText = Array.isArray(params.messageText)
+        ? selectSequentiallyFrom(params.messageText)
+        : params.messageText;
+
+    const urlParameters = {
+        REFPVID: params.pageviewId,
+        INTCMP: params.campaignCode,
+    };
+
+    const linkUrl = `${params.linkUrl}?${constructQuery(urlParameters)}`;
+
+    const renderedBanner = template(messageTemplate, {
+        linkHref: linkUrl,
+        messageText,
+        buttonCaption: params.buttonCaption,
+        colourClass,
+        arrowWhiteRight: inlineSvg('arrowWhiteRight'),
+        paypalLogoSrc: paypalAndCreditCardImage,
+    });
+
+    const messageShown = new Message(messageCode, {
+        pinOnHide: false,
+        siteMessageLinkName: 'membership message',
+        siteMessageCloseBtn: 'hide',
+        siteMessageComponentName: params.campaignCode,
+        trackDisplay: true,
+        cssModifierClass: colourClass,
+    }).show(renderedBanner);
+
+    if (messageShown) {
+        recordInteraction(params.interactionOnMessageShow);
+
+        mediator.emit('membership-message:display');
```

Comment:
This is being listened for in [`membership-engagement-banner-tests`](https://github.com/guardian/frontend/blob/master/static/src/javascripts-legacy/projects/common/modules/experiments/tests/membership-engagement-banner-tests.js#L43)